contracts/neutron_interchain_txs/src/contract.rs

```diff
@@ -251,21 +255,17 @@ fn execute_delegate(
 }
 
 fn execute_undelegate(
-    mut deps: DepsMut,
+    mut deps: DepsMut<InterchainQueries>,
     env: Env,
     interchain_account_id: String,
     validator: String,
     amount: u128,
     denom: String,
     timeout: Option<u64>,
-) -> StdResult<Response<NeutronMsg>> {
+) -> NeutronResult<Response<NeutronMsg>> {
     // contract must pay for relaying of acknowledgements
     // See more info here: https://docs.neutron.org/neutron/feerefunder/overview
-    let fee = IbcFee {
-        recv_fee: vec![],
-        ack_fee: vec![CosmosCoin::new(1000u128, "stake")],
-        timeout_fee: vec![CosmosCoin::new(1000u128, "stake")],
-    };
+    let fee = query_min_ibc_fee(deps.as_ref())?.min_fee;
```

Comment:
I mean this can probably mislead our users, cause we need to pick one denom out of min_fee response?

Reply:
+1 here

Reply:
but it is already implemented

Reply:
My bad missed it. Why the conversation isn't resolved tho

Reply:
the resolving of conversation has @NeverHappened (pun intended)

Reply:
wait, but it is not implemented correctly

Reply:
let's write a test for it maybe then?

Reply:
uhhhh up to you, but keep in mind it is just an example contract